File: Redistribute_Characters_to_Make_All_Strings_Equal.java
class Solution {
    public boolean makeEqual(String[] words) {
        Map<Character, Integer> chars = new HashMap<>();

        for (String word : words) {
            for (int i = 0; i < word.length(); i++) {
                chars.put(word.charAt(i), 1 + chars.getOrDefault(word.charAt(i), 0));
            }
        }

        int divider = words.length;

        for (int freq : chars.values()) {
            if (freq % divider != 0) {
                return false;
            }
        }

        return true;        
    }
}

File: Redistribute_Characters_to_Make_All_Strings_Equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        chars = {}

        for word in words:
            for i in range(len(word)):
                chars[word[i]] = 1 + chars.get(word[i], 0)
        
        divider = len(words)

        for freq in chars.values():
            if freq % divider != 0:
                return False
        
        return True